File: web/lib/schemas/teams.ts
import { z } from "zod";
import { userSchema } from "./user";
import { trainingCenterSchema } from "./training-center";

export const teamSchema = z.object({
  id: z.string(),
  name: z.string().min(1, { message: "Name must be required" }),
  coach_id: z.string(),
  center_id: z.string(),
  coach: userSchema,
  center: trainingCenterSchema,
  created_at: z.number(),
  updated_at: z.number(),
});

export type Team = z.TypeOf<typeof teamSchema>;

export const teamCreateSchema = teamSchema.pick({
  name: true,
  coach_id: true,
  center_id: true,
});

export type TeamCreate = z.TypeOf<typeof teamCreateSchema>;

export const teamUpdateSchema = teamSchema.pick({
  id: true,
  name: true,
  coach_id: true,
  center_id: true,
});

export type TeamUpdate = z.TypeOf<typeof teamUpdateSchema>;

File: api/internal/common/schemas/teams.py
from typing import List, Optional
from pydantic import BaseModel, Field

from internal.common.schemas.common import ListResponse, PagingRequest
from internal.common.schemas.training_center import CenterResponse
from internal.common.schemas.user import UserInfo
from internal.common.types import ID


class TeamInfo(BaseModel):
    coach_id: Optional[ID] = Field(None, description="Coach ID")
    center_id: Optional[ID] = Field(None, description="Training center ID")
    name: str = Field("", description="Team's name")


class CreateTeamRequest(TeamInfo):
    ...


class UpdateTeamRequest(TeamInfo):
    ...


class TeamResponse(TeamInfo):
    id: ID = Field(..., description="ID")
    coach: Optional[UserInfo] = Field(None, description="Coach info")
    center: Optional[CenterResponse] = Field(None, description="Center info")
    created_at: int = Field(..., description="Team created at")
    updated_at: int = Field(..., description="Team updated at")


class ListTeamRequest(PagingRequest):
    ...


class ListTeamResponse(ListResponse):
    teams: List[TeamResponse] = Field(..., description="List teams")
